jbpm-wb-integration/jbpm-wb-integration-backend/src/main/java/org/jbpm/workbench/wi/backend/server/workitem/WorkbenchRepositoryEventListener.java

```diff
@@ -160,12 +158,14 @@ public void onServiceTaskDisabled(RepoData service) {
     }
 
     @Override
-    public void onServiceTaskInstalled(RepoData service, String target, List<String> parameters) {
-        
+    public void onServiceTaskInstalled(RepoData service, String target, List<String> parameters, String branchName) {
         try {
-            
+            // default to master
+            if(branchName == null) {
+                branchName = "master";
+            }
             boolean installPomDepds = repositoryStorage.loadConfiguration().getInstallPomDeps();
-            Path path = PathFactory.newPath("pom.xml", REPOSITORY_PREFIX + target);
+            Path path = PathFactory.newPath("pom.xml", "default://" + branchName + "@" + target);
```

Comment:
I'd create a method for path building, because it is used in more places which might be prone to error when changing

Reply:
In fact also the null-check could be probably performed in the new method